### ISSUE TITLE
Fix bugs discovered by Envoy's fuzzers on bugs.chromium.org

### ISSUE THREAD

**Issue by htuch** [envoyproxy/envoy#4709]
Our fuzzers run continuously on ClusterFuzz and bugs are embargoed for 90 days. We have a significant backlog and we've allowed some of the non-critical fuzz bugs (functional vs. security related) to age out past the embargo. These can be seen at:

https://bugs.chromium.org/p/oss-fuzz/issues/list?can=2&q=envoy

This GH issue is intended to remain open for anyone interested in tackling an issue here. Most of these are great starter bugs, usually requiring a small PR to fix configuration validation or the integration test framework.

**Comment by sterchelen** [comment]
Hi, 

As a beginner, I can handle the 9960 issue: _"Stack-overflow in google::protobuf::TextFormat::Parser::ParserImpl::SkipFieldValue"_

**Comment by htuch** [comment]
Thanks @sterchelen, PRs welcome, please tag me on the PR.

**Comment by sterchelen** [comment]
Hi @htuch 
I've downloaded the test case for this fuzz issue ==> [here](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=9960&q=envoy&colspec=ID%20Type%20Component%20Status%20Proj%20Reported%20Owner%20Summary)

It seems very strange regarding its format, some double quotes are not closed and strange field name are created. Is that a normal behavior ?

**Comment by htuch** [comment]
@sterchelen the text protos are generated by https://github.com/google/libprotobuf-mutator, so should be validly formed. That said, this particular bug is a tricky one, as it probably depends on protobuf library internals and might be a corner case where we would not be able to solve Envoy-side. Do you want to try some other bug, e.g. https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=10130&q=envoy&colspec=ID%20Type%20Component%20Status%20Proj%20Reported%20Owner%20Summary?

**Comment by sterchelen** [comment]
All right, 
Yep, I will take it, the 10130 issue.

**Comment by sterchelen** [comment]
@htuch , I'm wondering how you debug envoy from the fuzzers side ? Do you have any documentation about that (debugging in general may be) ?

**Comment by htuch** [comment]
If you have a corpus entry, you can put this in the corpus directory for the fuzz test and debug using the same techniques as normal Envoy gtest, e.g. https://github.com/envoyproxy/envoy/tree/master/bazel#testing-envoy-with-bazel (and the sections below on gdb, stack traces, etc.). For oss-fuzz Chromium bugs, you should always have this corpus entry, so that's pretty much all. For debugging the build and fuzzers in the Docker environment, the oss-fuzz GH has more info, but I don't think you need that.

**Comment by sterchelen** [comment]
@htuch Thank you. I resolved to use gdb with envoy ! 
I got one question regarding the 10130 issue. I'm wondering if we have a limitation on header size. In fact, when we exceed 65536 bytes the test fail ! I saw that we are using nghttp2 but can't find any limit on header size.
Any idea ?

**Comment by twghu** [comment]
Fixed [OSS-Fuzz#22106](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=22106) with https://github.com/envoyproxy/envoy/pull/16405

**Comment by twghu** [comment]
Added fix for [OSS-Fuzz#30088](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=30088&q=label%3AReproducible%20envoy&can=2) with [#16405](https://github.com/envoyproxy/envoy/pull/16405)

**Comment by twghu** [comment]
Working on [OSS-Fuzz#30119](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=30119)

**Comment by twghu** [comment]
Reviewing [Fuzz Test 25777|https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=25777&q=label%3AReproducible%20envoy&can=2]

**Comment by ME-ON1** [comment]
@htuch Hi i was looking to get started on these issues and was looking on this issue https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=34875&can=2&q=envoy but couldn't get the idea from where to start !! 
Would love to get some starting pointers on this issue !

**Comment by twghu** [comment]
@ME-ON1 Hi Tarun, can you DM me on Slack/Envoy. 

To get started, have a read of 
https://github.com/envoyproxy/envoy/tree/main/test/fuzz#running-fuzz-tests-locally
and
https://opensource.googleblog.com/2021/02/writing-fuzz-tests-with-ease-using-bazel.html

**Comment by twghu** [comment]
NOTE in https://github.com/envoyproxy/envoy/tree/main/test/fuzz#running-fuzz-tests-locally 

`bazel run //test/common/common:base64_fuzz_test --config asan-fuzzer -- test/common/common/base64_corpus -runs=1000`

`-- test` not `--test`

**Comment by htuch** [comment]
Yeah, @twghu has some great advice to your question (and to anyone else looking to get going here).
